### PR TITLE
Move get-i18n bin to i18n-calypso and update `make translate`

### PR DESCRIPTION
**Pull request by Tug**
`bin/get-i18n` (as well as `sever/i18n`) which was used by `make translate` to extract strings from the build files into a unique `calypso-strings.php` file, has been moved into the new `i18n-calypso` module.

This PRs bumps the version of `i18n-calypso` to include this change and removes the tool from Calypso.
`make translate` is also updated to account for syntax changes in the cli.

## Testing instructions
Note: You will need the `xgettext` command to test this PR.

### 1. calypso-strings.php
Run the following commands to check that there is no regression in the way the POT file is generated (There will be a difference in the way strings are escaped in the `calypso-strings.php` so comparing the 2 won't help much)
- `git checkout master`
- `make install`
- `make translate`
- `xgettext calypso-strings.php --output=master.pot --keyword=__ --keyword=_n:1,2 --keyword=_x:1,2c --keyword=_nx:1,2,4c --language=php --from-code utf-8`
- `git checkout update/get-i18n-cli`
- `make install`
- `make translate`
- `xgettext calypso-strings.php --output=branch.pot --keyword=__ --keyword=_n:1,2 --keyword=_x:1,2c --keyword=_nx:1,2,4c --language=php --from-code utf-8`
- `vimdiff ./master.pot ./branch.pot`
- Check that there are no differences in the generated strings (there will be a few differences in the comments indicating the location of the strings in `calypso-strings.php`)

### 2. Localized dates (Review of https://github.com/Automattic/wp-calypso/pull/5278)
- Start calypso `make run`
- Switch to a translated non english locale (i.e 'es' or 'fr')
- Check that localized dates are still localized. For example, the date on hover over a post human date in the reader.

## Reviews
- [x] Code
- [x] Product 1.
- [x] Product 2.

Test live: https://calypso.live/?branch=update/get-i18n-cli

### CODE REVIEW

File: server/i18n/test/.gitignore [hunk 1]
```diff
@@ -1,2 +0,0 @@
-output.php
```
Comment (drewblaisdell) [inline server/i18n/test/.gitignore]: All of us will probably have a file at `server/i18n/test/output.php` that will now appear to be untracked with no explanation. Do you know of a solution to this?

**Comment by drewblaisdell** [comment]
I'm not seeing the same output in this branch vs. master. The strings from this branch appear to be missing miscellaneous strings at the top and bottom of the master output, including moment.js format strings. You can see this with `vimdiff`:

```
vimdiff calypso-strings-branch.txt calypso-strings-master.txt
```

I renamed the files to .txt to upload to GitHub.

[calypso-strings-master.txt](https://github.com/Automattic/wp-calypso/files/307781/calypso-strings-master.txt)
[calypso-strings-branch.txt](https://github.com/Automattic/wp-calypso/files/307782/calypso-strings-branch.txt)

File: server/i18n/test/.gitignore (hunk 1 above)
Comment (Tug) [inline server/i18n/test/.gitignore]: You're right. No I haven't really thought about it and I don't see any practical solution except telling devs to remove this file manually :s

**Comment by Tug** [comment]
The last version of `i18n-calypso` has the changes from #5278. I included the commit which removed a webpack plugin in this PR so we don't have to review this upgrade twice.

I agree that there are some strings missing now in `calypso-strings.php`, especially the first few ones:
```
__( 'All Sites' ),
__( '%s ‹ Reader' ),
__( 'Post not found' ),
__( 'Post Not Found' ),
__( 'Following' ),
__( 'Followed Sites' ),
__( 'Follow' ),
__( 'Following' ),
...
_x( 'Status', 'Filter group label for tabs' ),
_x( 'Author', 'Filter group label for segmented' ),
```

I'm investigating this.

**Comment by Tug** [comment]
There was a bug in the cli tool in the way it reads the input files (it was removing the first 2 files when we don't use the -i argument), I have fixed it in `i18n-calypso` https://github.com/Automattic/i18n-calypso/commit/36cc86d7d70bde53ca674e831003d7d554d9dd13 and released a new version.
I think it's ready for another review :)

**Comment by scruffian** [comment]
I am still seeing some differences between master and this branch in `calypso-strings.php`. Most differences are related to escaping only (`'` vs `"`). I think it would be better to make these changes in a different PR.

**Comment by Tug** [comment]
This has been fixed in the last version of `i18n-calypso` here https://github.com/Automattic/i18n-calypso/commit/db815e50ebc06204f5b1b873ff471be707e70ff4
Testing instructions have been updated :)

**Comment by scruffian** [comment]
LGTM 👍

**Pull request merged**